tests/Unit/NumericalAlgorithms/Spectral/Test_Basis.cpp

```diff
@@ -0,0 +1,101 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <array>
+#include <catch.hpp>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Matrix.hpp"
+#include "NumericalAlgorithms/Spectral/Basis.hpp"
+#include "Parallel/Printf.hpp"
+#include "Utilities/Blas.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+namespace {
+
+template <Basis::Polynomials Poly, Basis::Quadrature Quad>
+void test_linear_filter_matrix() {
+  for (size_t n = 2; n < 10; ++n) {
+    const Matrix& filter_matrix = Basis::linear_filter_matrix<
+      Poly, Quad>(n);
+    const Matrix& grid_points_to_spectral_matrix =
+        Basis::grid_points_to_spectral_matrix<Poly, Quad>(n);
+    const DataVector& collocation_points = Basis::collocation_points<
+      Poly, Quad>(n);
+    DataVector u(n);
+    for (size_t s = 0; s < n; ++s) {
+      u[s] = exp(collocation_points[s]);
```

Comment:
We have vectorized maths operations on DataVectors, so lines 25-28 can be replaced with `const DataVector u = exp(collocation_points)`